dspback/routers/discovery.py

```diff
@@ -84,50 +84,19 @@ async def search(
                 }
             },
         )
-        stages[0]['$search']['count'] = {'type': 'total'}
-        stages.append({'$project': {'meta': "$$SEARCH_META",
-                                    '_id': 0,
-                                    '@context': 1,
-                                    'repository_identifier': 1,
-                                    'url': 1,
-                                    '@type': 1,
-                                    'provider': 1,
-                                    'name': 1,
-                                    'description': 1,
-                                    'keywords': 1,
-                                    'creator': 1,
-                                    'funding': 1,
-                                    'temporalCoverage': 1,
-                                    'spatialCoverage': 1,
-                                    'license': 1,
-                                    'datePublished': 1,
-                                    'dateCreated': 1,
-                                    'relations': 1,
-                                    'legacy': 1,
-                                    'clusters': 1,
-                                    'score': 1
-                                    }
-                       })
 
     if term:
         stages[0]['$search']['highlight'] = {'path': search_paths}
         # get only results which meet minimum relevance score threshold
         score_threshold = get_settings().search_relevance_score_threshold
         stages.append({'$match': {'score': {'$gt': score_threshold}}})
-        for stage in stages:
-            if '$project' in stage:
-                stage['$project']['highlights'] = {'$meta': 'searchHighlights'}
-                break
 
-    results = await request.app.db[get_settings().mongo_database]["discovery"].aggregate(stages).to_list(pageSize)
+    # TODO: implement for other search methods
+    # Insert a facet stage before pagination to extract the total count
+    stages.append({ "$facet": {"results": [{ "$skip": (pageNumber - 1) * pageSize }, { "$limit": pageSize }], "totalCount": [{ "$count": 'count'}]}})
```

Comment:
NICE. I too was trying to use facet stage but could not make it to work. I assume this gives the correct record count.

Reply:
Sounds good. Once you are done making the changes I can help refactor if necessary.